dora/core/server/master/src/main/java/alluxio/master/scheduler/MembershipManagerWorkerProvider.java

```diff
@@ -38,11 +41,22 @@ public MembershipManagerWorkerProvider(MembershipManager membershipMgr,
                                          FileSystemContext context) {
     mMembershipManager = membershipMgr;
     mContext = context;
+    if (context != null && context.getClusterConf() != null) {
```

Comment:
The previous implementation wasn't consistent with the client implementation where USER_DYNAMIC_CONSISTENT_HASH_RING_ENABLED is checked to determine if dead workers are included or not. 

I'm open to revoke this change though but i just feel the current status is not consistent.